src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts

```diff
@@ -142,19 +140,24 @@ export class AssessmentDashboardComponent implements OnInit {
                     }
                     this.numberOfCorrectionRounds = this.numberOfAssessmentsOfCorrectionRounds.length;
 
-                    this.numberOfComplaints = this.stats.numberOfComplaints;
-                    this.numberOfOpenComplaints = this.stats.numberOfOpenComplaints;
-                    this.numberOfAssessmentLocks = this.stats.numberOfAssessmentLocks;
-                    this.totalNumberOfAssessmentLocks = this.stats.totalNumberOfAssessmentLocks;
                     const tutorLeaderboardEntry = this.stats.tutorLeaderboardEntries?.find((entry) => entry.userId === this.tutor.id);
                     this.sortService.sortByProperty(this.stats.tutorLeaderboardEntries, 'points', false);
                     if (tutorLeaderboardEntry) {
                         this.numberOfTutorAssessments = tutorLeaderboardEntry.numberOfAssessments;
-                        this.numberOfTutorComplaints = tutorLeaderboardEntry.numberOfTutorComplaints;
+                        this.complaints = new AssessmentDashboardInformationEntry(
+                            this.stats.numberOfComplaints,
+                            tutorLeaderboardEntry.numberOfTutorComplaints,
+                            this.stats.numberOfComplaints - this.stats.numberOfOpenComplaints,
+                        );
                     } else {
                         this.numberOfTutorAssessments = 0;
-                        this.numberOfTutorComplaints = 0;
+                        this.complaints = new AssessmentDashboardInformationEntry(
+                            this.stats.numberOfComplaints,
+                            0,
+                            this.stats.numberOfComplaints - this.stats.numberOfOpenComplaints,
+                        );
                     }
+                    this.assessmentLocks = new AssessmentDashboardInformationEntry(this.stats.totalNumberOfAssessmentLocks, this.stats.numberOfAssessmentLocks);
```

Comment:
The `loadAll()` method in general is a bit of a monster, did you thought about somehow splitting it up? just an idea...